src/service/grpc.cpp

```diff
@@ -40,6 +41,57 @@ grpc::ServerUnaryReactor *Grpc::CreateCollection(
 	return reactor;
 }
 
+grpc::ServerUnaryReactor *Grpc::RetrieveCollection(
+	grpc::CallbackServerContext *context, const gk::v1::RetrieveCollectionRequest *request,
+	gk::v1::Collection *response) {
+	auto *reactor = context->DefaultReactor();
+
+	try {
+		auto collection = datastore::RetrieveCollection(request->id());
+		map(collection, response);
+	} catch (const err::DatastoreCollectionNotFound &) {
+		reactor->Finish(grpc::Status(grpc::StatusCode::NOT_FOUND, "Document not found"));
+		return reactor;
+	} catch (...) {
+		reactor->Finish(grpc::Status(grpc::StatusCode::UNAVAILABLE, "Failed to retrieve data"));
+		return reactor;
+	}
+
+	reactor->Finish(grpc::Status::OK);
+	return reactor;
+}
+
+grpc::ServerUnaryReactor *Grpc::UpdateCollection(
+	grpc::CallbackServerContext *context, const gk::v1::UpdateCollectionRequest *request,
+	gk::v1::Collection *response) {
+	auto *reactor = context->DefaultReactor();
+
+	try {
+		auto collection = datastore::RetrieveCollection(request->id());
+		bool hasUpdates = false;
+		if (request->has_name()) {
```

Comment:
Probably better to check if there's anything to update and throw an error if not before doing a DB lookup since the client shouldn't have send such a request.